src/keysBuilder/buildKeys.ts

```diff
@@ -9,5 +10,11 @@ export function buildKeys(config: Config) {
   const scopeToKeys = mergeDeep({}, template.scopeToKeys, ts.scopeToKeys);
   const fileCount = template.fileCount + ts.fileCount;
 
+  if (config.unflat) {
+    for (const [_, scopeKeys] of Object.entries(scopeToKeys)) {
+      checkForProblematicUnflatKeys(scopeKeys as object);
```

Comment:
I think we can get a more readable and easily implemented function if we use the flattened file.
We can iterate over the keys and check for problematic keys as follows:
When given this sorted flat JSON:
```json
{
 "a": "missing",
 "a.b": "missing",
}
```
All we need to do is iterate over each key and check if the next one is starting with it followed by a dot.

Reply:
At that stage the files are not yet created/updated and if you are using unflat setting then those files will be missing the problematic keys. Here the existing in-memory map is used which is the representation of the (flat) JSON in the end.
Can you give me some more info? Actually the implementation of `checkForProblematicUnflatKeys` is just using the approach you mention.

Reply:
Yes, I can see now that we are talking about the same approach 🙂.
Just a couple of small fixes to make this clearer:

1. Add `ScopeMap` type to `scopeToKeys` and remove the casting to object.
2. Use `Object.values` instead of `entries`